ocaml/xapi/xapi_vm.ml

```diff
@@ -249,21 +249,30 @@ let start ~__context ~vm ~start_paused ~force =
   let default_value = match Xapi_xenops.firmware_of_vm vmr with
     | Bios -> Vm_platform.fallback_device_model_default_value
     | Uefi _ -> Vm_platform.fallback_device_model_default_value_uefi in
-  let platform = vmr.API.vM_platform
+  let platform = ref( vmr.API.vM_platform
```

Comment:
Why do we need a reference here? Isn't it possible to determine the value of platform before using it later?